### PR TITLE
Add torch.backends.openmp.is_available(); fix some cmake messages

### PR DESCRIPTION
**Pull request by ssnl**
1. add `torch.backends.openmp.is_available()`
2. Improve various `cmake` outputs
3. Fix LDFLAGS not respected by `caffe2_pybind11_state_*` targets
4. Fix `MKL` warning message, and QUIET flag.
5. Fix various typos

### CODE REVIEW

File: cmake/Dependencies.cmake [hunk 1]
```diff
@@ -113,22 +113,18 @@ if (NOT BUILD_ATEN_MOBILE)
...
       set(USE_BLAS 0)
+      set(BLAS "" CACHE STRING "Selected BLAS library")
+    else()
+      set(BLAS BLAS_INFO CACHE STRING "Selected BLAS library")
     endif()
   endif()
 
   if (MKL_FOUND)
     ADD_DEFINITIONS(-DTH_BLAS_MKL)
-    if(NOT MKL_INCLUDE_DIR)
```
Comment (SsnL) [inline cmake/Dependencies.cmake]: Now in `FindMKL.cmake` having `MKL_INCLUDE_DIR` is a requirement for `MKL_FOUND` to be set, so this message is never printed. I moved this message to `FindMKL.cmake`.

**Comment by ezyang** [comment]
cc @JerryShih

**Comment by JerryShih** [comment]
@SsnL 
I think I do the similar things in: https://github.com/pytorch/pytorch/pull/15820

I also show the explicit openmp status message for "pytorch". The pytorch just use openmp by default, but the openmp is not turned on by default[[1]](https://github.com/pytorch/pytorch/blob/5ebf4cd4e3c4fa86d7c66d9115034cbc2a919da6/CMakeLists.txt#L128).

The mac platform ci log in this pr still show that it can't find the openmp configuration.
```
Could NOT find OpenMP_C (missing: OpenMP_C_FLAGS OpenMP_C_LIB_NAMES) 
Could NOT find OpenMP_CXX (missing: OpenMP_CXX_FLAGS OpenMP_CXX_LIB_NAMES) 
Could NOT find OpenMP (missing: OpenMP_C_FOUND OpenMP_CXX_FOUND) 
```

**Comment by SsnL** [comment]
@JerryShih I think ATen tries to build with OMP regardless of the value `WITH_OPENMP` has. I have successfully built PyTorch w/ OMP on CI. But I haven't gotten caffe 2 working. I suspect some `target_link_libraries` in `caffe2/CMakeList.txt` is needed, as you did in your PR. Should I close this one in favor of your PR?

**Comment by JerryShih** [comment]
@SsnL 
It sounds good. Hope my pr could resolve all problem. I am waiting for reviewing. The ci looks good.

**Comment by SsnL** [comment]
@JerryShih If you permit, I can push the OpenMP test I have in this PR (the `torch._C.has_openmp` flag) to your PR branch. :)

**Comment by ezyang** [comment]
Please do :)

**Comment by SsnL** [comment]
@ezyang I have some other fixes/changes in this PR that I would love to merge. So I just purged AppleClang parts from this. Now this PR is like doing four different things but I hope that you can still review them as they are small changes. :)

**Review by ezyang — APPROVED** [review]

**Comment by SsnL** [comment]
@ezyang build failures are fixed in #16565

**Review by soumith — APPROVED** [review]

**Pull request closed without merging**